File: GameCore/src/game/core/ITrackPiece.java
package game.core;

/**
 * Фигура, которая может делать последовательность коротких ходов.
 *
 * @author <a href="mailto:vladimir.romanov@gmail.com">Romanov V.Y.</a>
 */
public interface ITrackPiece {
    /**
     * Есть ли правильный ход из клетки square.
     *
     * @param square - проверяемая клетка
     * @return есть ли правильных ход из клетки.
     */
    boolean hasCorrectMoveFrom(Square square);
}

File: GameCore/src/game/core/ITrackPiece.kt
package game.core

/**
 * Фигура, которая может делать последовательность коротких ходов.
 *
 * @author [Romanov V.Y.](mailto:vladimir.romanov@gmail.com)
 */
interface ITrackPiece {
    /**
     * Есть ли правильный ход из клетки square.
     *
     * @param square - проверяемая клетка
     * @return есть ли правильных ход из клетки.
     */
    fun hasCorrectMoveFrom(square: Square?): Boolean
}